File: src/ProjectR.Scripting/content/scripts/spells/lua/ScarletSwordSlash.lua
function GetTargetType()
    return Single
end

function GetName()
  return "Scarlet Sword Slash"
end

function GetDescription()
  return "Slash your scarlet sword at the enemy.\nUseful on high DEF enemies."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .52
end

function GetDelay()
  return .4
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  local damage = 3.5 * aAD - .4375 * dDEF
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/ScarletSwordSlash.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class ScarletSwordSlash : SpellScriptBase
    {
        public override string Name { get { return "Scarlet Sword Slash"; } }
        public override string Description { get { return "Slash your scarlet sword at the enemy.\nUseful on high DEF enemies."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 52; } }
        public override double Delay { get { return .4; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = 3.5 * aAD - .4375 * dDEF;
            DealDamage(damage);
        }
    }
}